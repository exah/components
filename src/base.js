import React from 'react'
import { identity } from '@exah/utils'

export function base ({
  use: defaultComp = 'div',
  filter = identity
} = {}) {
  const Base = React.forwardRef(({ as = defaultComp, use: Comp = as, ...rest }, ref) => (
    <Comp ref={ref} {...filter(rest, Comp)} />
  ))

  Base.displayName = 'Base'
  return Base
}
