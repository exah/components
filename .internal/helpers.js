import renderer from 'react-test-renderer'
import expect from 'expect'

export const renderJSON = (element) =>
  renderer.create(element).toJSON()

export const anyClassName = expect.stringMatching(/^\./)
