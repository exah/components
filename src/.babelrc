{
  "env": {
    "modules": {
      "plugins": [
        "transform-es2015-modules-commonjs",
        [ "emotion", { "hoist": true } ]
      ]
    },
    "production": {
      "plugins": [
        [ "emotion", { "hoist": true } ]
      ]
    },
    "development": {
      "plugins": [
        ["emotion", { "sourceMap": true }]
      ]
    }
  },
  "presets": [
    [ "env", {
      "useBuiltIns": true,
      "modules": false,
      "loose": true
    } ]
  ],
  "plugins": [
    "transform-react-jsx",
    "transform-class-properties",
    [ "transform-object-rest-spread", { "useBuiltIns": true } ]
  ]
}
