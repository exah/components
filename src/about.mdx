---
name: Getting Started
route: /
order: 1
---

# Getting Started
