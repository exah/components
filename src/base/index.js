import Base, { createBase } from 'react-styled-base'

export {
  Base,
  createBase
}
