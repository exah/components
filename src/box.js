import styled from '@emotion/styled'
import { getPropTypes } from 'pss/prop-type'
import { combineStyles, box, boxStyle, themePath } from 'pss'
import { base, omit } from './utils'

const styles = combineStyles(
  boxStyle,
  box
)

const boxDefaultStyles = {
  font: 'inherit',
  color: 'inherit',
  background: 'transparent',
  border: 0,
  margin: 0,
  padding: 0
}

const BaseBox = base({
  name: 'Box',
  filter: omit(styles.props)
})

const Box = styled(BaseBox)(
  { boxSizing: 'border-box' },
  themePath('Box', boxDefaultStyles),
  styles
)

Box.displayName = 'Box'

Box.propTypes = {
  ...BaseBox.propTypes,
  ...getPropTypes(styles)
}

export {
  Box
}
