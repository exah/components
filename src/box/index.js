import {
  combineStyles,
  border,
  boxSelfAlignment,
  boxStyle,
  clearFix,
  colors,
  cursor,
  display,
  flex,
  float,
  hideOn,
  opacity,
  order,
  outline,
  position,
  positionOffset,
  radius,
  ratio,
  sizes,
  space,
  transform,
  transition,
  zIndex
} from 'pss'

import PropTypes from 'prop-types'
import styled from '@emotion/styled'
import { Base } from '../base'

const BorderBox = styled(Base)(
  { boxSizing: 'border-box' },
  (props) => props.initialCSS
)

BorderBox.displayName = 'BorderBox'
BorderBox.propTypes = {
  ...Base.propTypes,
  initialCSS: PropTypes.object
}

const styles = combineStyles(
  border,
  boxSelfAlignment,
  boxStyle,
  clearFix,
  colors,
  cursor,
  display,
  flex,
  float,
  hideOn,
  opacity,
  order,
  outline,
  position,
  positionOffset,
  radius,
  ratio,
  sizes,
  space,
  transform,
  transition,
  zIndex
)

const Box = styled(BorderBox)(styles)

Box.propTypes = {
  ...BorderBox.propTypes,
  ...styles.propTypes
}

export {
  Box,
  BorderBox
}
