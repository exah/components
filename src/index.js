export { Box } from './box'
export { Button } from './button'
export { Feed } from './feed'
export { Flex } from './flex'
export { FlexGrid } from './flex-grid'
export { Grid } from './grid'
export { Input } from './input'
export { Image } from './image'
export { Layout } from './layout'
export { Link } from './link'
export { List } from './list'

export {
  MatchMediaContext,
  MatchMediaProvider,
  MatchMediaConsumer,
  useMatchMedia,
  useMatchMediaContext,
  withMatchMedia
} from './match-media'

export { Text } from './text'

export {
  ThemeDefaults,
  ThemeProvider,
  useTheme
} from './theme'

export { base } from './utils'
