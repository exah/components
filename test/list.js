import React from 'react'
import { renderJSON, anyClassName } from '../.internal/helpers'
import { List } from '../src'

describe('List', () => {
  test('snapshot', () => {
    const result = renderJSON(
      <List>
        <List.Item>Item</List.Item>
      </List>
    )

    expect(result).toMatchSnapshot()
  })

  test('as selector', () => {
    expect(`${List}`).toEqual(anyClassName)
    expect(`${List.Item}`).toEqual(anyClassName)
  })
})
