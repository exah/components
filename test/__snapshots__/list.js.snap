// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List snapshot 1`] = `
.emotion-3 {
  box-sizing: border-box;
  font: inherit;
  color: inherit;
  background: transparent;
  border: 0;
  margin: 0;
  padding: 0;
  list-style: none;
}

.emotion-0 {
  box-sizing: border-box;
  font: inherit;
  color: inherit;
  background: transparent;
  border: 0;
  margin: 0;
  padding: 0;
}

<ul
  className="emotion-2 emotion-3 emotion-4"
>
  <li
    className="emotion-0 emotion-1"
  >
    Item
  </li>
</ul>
`;
